### PR TITLE
Port test_encode to Python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3419

Builds on #812, so review that first.

### CODE REVIEW

File: src/allmydata/uri.py [hunk 1]
```diff
@@ -13,8 +13,9 @@
 
 from future.utils import PY2
 if PY2:
-    # Don't import bytes, to prevent leaks.
-    from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, dict, list, object, range, str, max, min  # noqa: F401
+    # Don't import bytes or str, to prevent leaks.
```
Comment (sajith) [inline src/allmydata/uri.py:16]: What are those leaks?

File: src/allmydata/test/no_network.py [hunk 2]
```diff
@@ -308,6 +325,7 @@ def __init__(self, basedir, num_clients, num_servers,
             d.addCallback(lambda c: self.clients.append(c))
 
             def _bad(f):
+                print(f)
```
Comment (sajith) [inline src/allmydata/test/no_network.py:328]: This must be a stray `print`?

**Review by sajith — APPROVED** [review]
Looks good to me.  Thank you!

Please merge when you're ready.

File: src/allmydata/uri.py (hunk 1 above)
Comment (itamarst) [reply src/allmydata/uri.py]: I clarified the comment.

**Comment by itamarst** [comment]
This passed on last run, and the changes are trivial (comment tweak), so this looks like intermittent failures.

**Pull request merged**